src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/WritingPrimitives.cs

```diff
@@ -0,0 +1,312 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Serializer;
+
+internal static class WritingPrimitives
```

Comment:
nit: `WritingPrimitives` -> `ProtobufSerializer`

Other options `ProtobufSerializationHelper`, `ProtobufPrimitiveSerializer`

More or less what I'm saying is "WritingPrimitives" sucks as a name 😄